final_task/rss_parser/Output_functions.py

```diff
@@ -0,0 +1,227 @@
+            # go with soup 'img'
+            for image in range(getting_num_of_news(the_feed, num_of_news)):
+                pack_of_images_links.append(item.get('media_content', '')[0].get('url', ''))
+        else:
+            try:
+                pack_of_img.append(soup.find('img'))
+                if soup.find('img') is not None:
+                    pack_of_images_links.append(pack_of_img[num]['src'])
+                else:
+                    pack_of_images_links.append('No image')
+            except IndexError:
+                logging.warning("Some problems appeared, solving them!")
+                for img in soup.find_all('img'):
+                    if img.get('src') != '':
+                        pack_of_images_links.append(img.get('src'))
+                    else:
+                        pack_of_images_links.append(num)
+    logging.info("Got some images!")
+    return pack_of_images_links
+
+
+def getting_alt_text(the_feed, num_of_news):
+    """
+    getting alternative text for images
+    :param the_feed:
+    :param num_of_news:
+    :return:
+    """
+    pack_of_alts = []
+    logging.info("Getting alternative text for images!")
+    for num, item in enumerate(the_feed.entries):
+        if num == getting_num_of_news(the_feed, num_of_news):  # ограничение количества отображаемых новостей
+            break
+        soup = BeautifulSoup(str(item), "lxml")
+        for img in soup.find_all('img'):
+            if img is None:
+                pack_of_alts.append(num)
+            else:
+                try:
+                    pack_of_alts.append(img['alt'])
+                except KeyError:
+                    logging.warning("Solving problems with alternative text!")
+                    pack_of_alts.append(num)
+    logging.info("Got some alternative text.")
+    return pack_of_alts
+
+
+def getting_pack_of_news(the_feed, num_of_news=None):
+    """
+    Creating full novelty
+    1. Adding list of images, correct it if some duplicates are there
+    2. Adding list of alternative texts, correct it if some duplicates are there
+    3. Some problems can occur if there are no alternative text, solving them with changing list of alts
+    :param the_feed:
+    :param num_of_news:
+    :return:
+    """
+    if num_of_news > len(the_feed.entries):
+        print("You want to get more news than exist.")
+        print("Printing all news from the source!")
+
+    print("------------------------")
+    print("Source: ", printing_title(the_feed))
+    pack_of_news = []
+    pack_of_images_links = getting_images_links(the_feed, num_of_news)
+    pack_of_alts = getting_alt_text(the_feed, num_of_news)
+    # corrected_pack_of_images_links = [el for el, _ in groupby(pack_of_images_links)]
+    corrected_pack_of_alts = [el for el, _ in groupby(pack_of_alts)]
+
+    for num, item in enumerate(the_feed.entries):
+        if num == num_of_news:  # ограничение количества отображаемых новостей
```

Comment:
Instead of checking `num` on every iteration, you can just define the number of iterations. Something like:
```
for num, item in enumerate(the_feed.entries[;num_of_news]):
 ...
```
So you will not need this check at all
Remember, that iteration starts from 0

Reply:
Same for simmilar code above and below

Reply:
Thanks, fixed.